File: unity_code/Actions/SequenceAction.cs
using System.Collections.Generic;

namespace Assets.Scripts.Actions
{
    public class SequenceAction : IGameAction
    {
        private Queue<IGameAction> actionQueue;
        private IGameAction currentAction;

        public SequenceAction (params IGameAction[] sequencedActions)
        {
            actionQueue = new Queue<IGameAction>();

            foreach(var action in sequencedActions)
                actionQueue.Enqueue(action);

            currentAction = actionQueue.Dequeue();
        }

        public ActionResult Step (float deltaTime)
        {
            var result = currentAction.Step(deltaTime);

            if(result == ActionResult.Continue)
                return ActionResult.Continue;

            if(actionQueue.Count == 0)
                return ActionResult.Finished;

            currentAction = actionQueue.Dequeue();
            return ActionResult.Continue;
        }
    }
}

File: src/main/ts/actions/SequenceAction.ts

    import {IGameAction} from "./IGameAction";
    import {ActionResult} from "./ActionResult";
    export class SequenceAction implements IGameAction
    {
        private actionQueue:Array<IGameAction>;
        private currentAction:IGameAction;

        public constructor(sequencedActions:Array<IGameAction>)
        {
            this.actionQueue = [];

            for(let action of sequencedActions) {
                this.actionQueue.push(action);
            }

            this.currentAction = this.actionQueue.shift();
        }

        public step(deltaTime:number):ActionResult
        {
            let result = this.currentAction.step(deltaTime);

            if(result == ActionResult.Continue) {
                return ActionResult.Continue;
            }

            if(this.actionQueue.length == 0) {
                return ActionResult.Finished;
            }

            this.currentAction = this.actionQueue.shift();
            return ActionResult.Continue;
        }
    }